src/lib/parse-url.js

```diff
@@ -5,23 +5,36 @@
  * supporting a default value, so a simple split is sufficent.
  * @param {string} url
  */
-export default function parseUrl (url) {
+export default function parseUrl(url) {
```

Comment:
This function is also called server-side, and the logic currently works because there we pass it `NEXTAUTH_URL`. 

I think we should move that logic inside, hence whenever calling `parseUrl` it can automatically compute what the right **base URL** and **base path** are:
```ts
export default function parseBaseUrl(url) {
// ..
  if (process.env.NEXTAUTH_URL) {
    if (!process.env.NEXT_PUBLIC_NEXTAUTH_URL) {
      throw new Error("[next-auth] Trying to use a custom `NEXTAUTH_URL` without setting `NEXT_PUBLIC_NEXTAUTH_URL`, client-side redirects won't work")
```
Now that we're modifying this function, I think we should:
- rename it as it's not generic utility about URLs but is intended to compute the base URL
- `throw` in case expectations don't match so we're explicit about mis-configurations

Reply:
> I think we should move that logic inside, hence whenever calling parseUrl it can automatically compute what the right base URL and base path are

Yes, currently, it feels like it's dangling between both client and server.  

> throw in case expectations don't match so we're explicit about mis-configurations 

Agreed.